src/Microsoft.Diagnostics.Monitoring.WebApi/Utilities/StackUtilities.cs

```diff
@@ -20,15 +20,24 @@ internal enum StackFormat
 
     internal static class StackUtilities
     {
-        public static Models.CallStack TranslateCallStackToModel(CallStack stack, NameCache cache)
+        public static Models.CallStack TranslateCallStackToModel(CallStack stack, NameCache cache, bool methodNameIncludesGenericParameters = true)
```

Comment:
Some suggestions to simplify this a bit:
- Add a `TypeFormat` parameter to this method, make the caller decide up front which format it wants the type information for the model to be in.
- Remove `CallStackFrame.FullTypeArgs` and `CallStackFrame.FullParameterTypes`. Populate the existing `CallStackFrame.TypeArgs` and `CallStackFrame.ParameterTypes` based on the `TypeFormat` parameter.

Reply:
My understanding is that for Exceptions we do this translation when we receive an exception (not when the user hits `/exceptions`) - as a result we don't know whether they'll want JSON/text. We don't have that problem for `/stacks`, but without changing when we do this translation from the stack to the model so that it happens on-demand, I'm not sure how we would know the `TypeFormat`?

Reply:
Good point, I'll think about this some more and see if a unified model can still be done.